pyomo/gdp/plugins/chull.py

```diff
@@ -171,6 +178,33 @@ def __init__(self):
             Block:       self._transform_block_on_disjunct,
             }
 
+    def _add_local_vars(self, block, local_var_dict):
+        localVars = block.component('LocalVars')
+        if type(localVars) is Suffix:
+            for disj, var_list in iteritems(localVars):
+                if local_var_dict.get(disj) is None:
+                    local_var_dict[disj] = ComponentSet(var_list)
+                else:
+                    local_var_dict[disj].update(var_list)
+
+    def _get_local_var_suffixes(self, block, local_var_dict):
+        # You can specify suffixes on any block (dijuncts included). This method
+        # starts from a Disjunct (presumably) and checks for a LocalVar suffixes
+        # going up the tree, adding them into the dictionary that is the second
+        # argument.
+
+        # first look beneath where we are (there could be Blocks on this
+        # disjunct)
+        for b in block.component_data_objects(Block, descend_into=(Block),
+                                              active=True,
+                                              sort=SortComponents.deterministic):
+            self._add_local_vars(b, local_var_dict)
```

Comment:
Is there a possibility for suffix settings to conflict here? If so, maybe search upwards and then down, so that the "closest"  value wins?

Reply:
@qtothec, no, there can't be a conflict because you can't declare a variable as *not* local using this Suffix. So the order doesn't matter: If we find the variable anywhere, it's local. Otherwise, no.